src/utils/crypto/shouldSkipSetupEncryption.ts

```diff
@@ -9,14 +9,21 @@ Please see LICENSE files in the repository root for full details.
 import { MatrixClient } from "matrix-js-sdk/src/matrix";
 
 import { shouldForceDisableEncryption } from "./shouldForceDisableEncryption";
+import { asyncSome } from "../arrays.ts";
 
 /**
  * If encryption is force disabled AND the user is not in any encrypted rooms
  * skip setting up encryption
  * @param client
  * @returns {boolean} true when we can skip settings up encryption
  */
-export const shouldSkipSetupEncryption = (client: MatrixClient): boolean => {
+export const shouldSkipSetupEncryption = async (client: MatrixClient): Promise<boolean> => {
     const isEncryptionForceDisabled = shouldForceDisableEncryption(client);
-    return isEncryptionForceDisabled && !client.getRooms().some((r) => client.isRoomEncrypted(r.roomId));
+    const crypto = client.getCrypto();
+    if (!crypto) return true;
+
+    return (
+        isEncryptionForceDisabled &&
+        !(await asyncSome(client.getRooms(), ({ roomId }) => crypto.isEncryptionEnabledInRoom(roomId)))
```

Comment:
This seems unnecessarily slow, should it not be a Promise.race?

Reply:
I don't think so. `Promise.race` fulfils the first promise to settle which is not good for this scenario:
1. Two rooms, two promises: p1 & p2
2. If p1 settles first and the room is unencrypted => `Promise.race` returns `false`
3. If p2 settles first and the room is encrypted => `Promise.race` returns `true`

`asyncSome` stops to iterate when a promise is resolved.

Maybe thought about `Promise.any` and to reject the promises when the room is encrypted?